pkg/booleanpolicy/networkpolicy/networkpolicy.go

```diff
@@ -0,0 +1,31 @@
+package networkpolicy
+
+import (
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/booleanpolicy/augmentedobjs"
+)
+
+// GetNetworkPoliciesApplied creates an augmentedobj.NetworkPoliciesApplied object
+// based on a provided map of network policies.
+func GetNetworkPoliciesApplied(networkPolicies map[string]*storage.NetworkPolicy) *augmentedobjs.NetworkPoliciesApplied {
+	var hasIngress, hasEgress bool
+	for _, policy := range networkPolicies {
+		for _, policyType := range policy.GetSpec().GetPolicyTypes() {
+			hasIngress = hasIngress || policyType == storage.NetworkPolicyType_INGRESS_NETWORK_POLICY_TYPE
+			hasEgress = hasEgress || policyType == storage.NetworkPolicyType_EGRESS_NETWORK_POLICY_TYPE
+			if hasIngress && hasEgress {
+				return &augmentedobjs.NetworkPoliciesApplied{
+					MissingIngressNetworkPolicy: false,
+					MissingEgressNetworkPolicy:  false,
+					Policies:                    networkPolicies,
```

Comment:
Why attach all the net pols? Is the purpose of this func to only determine the _presence_ of any ingress and egress netpols, and re-evaluate again elsewhere which net pols are ingress and egress? We can elaborate this to be a list of ingress net pols and list of egress net pols?

Reply:
This is part of a ticket to display all network policies related to deployment in the UI. The UI part is not implemented yet, but the API returns already the extra info.

See: https://issues.redhat.com/browse/ROX-9919 (API) and https://issues.redhat.com/browse/ROX-9760 (UI)

Reply:
I see. Thanks for the clarification. The method `getNetworkPoliciesApplied(...)` is generating  `NetworkPoliciesApplied` object for a deployment, whereas, exported method `GetNetworkPoliciesApplied(...)` is simply constructing the final `NetworkPoliciesApplied` object by filling in the netpol type. We should bring it out in the func name: `GetNetworkPoliciesApplied` -> `GenNetworkPoliciesAppliedObj`

Reply:
Done